src/client/api/v1/get_balance.c

```diff
@@ -0,0 +1,170 @@
+      ret = -1;
+      free(code);
+      goto end;
+
+    } else if (strcmp(code, "not_found") == 0) {
+      printf("[%s:%d]: http code 404\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      free(code);
+      goto end;
+    }
+
+    free(code);
+  }
+
+  cJSON *data_obj = cJSON_GetObjectItemCaseSensitive(json_obj, key_data);
+  if (data_obj) {
+    // gets addr
+    char *addr = calloc(64, sizeof(char));
+    if ((ret = json_get_string(data_obj, key_addr, addr, 64)) != 0) {
+      printf("[%s:%d]: gets %s json addr failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      goto end;
+    }
+
+    if (strlen(addr) == 64) {
+      hex2bin(addr, res->addr + 1, IOTA_ADDRESS_BYTES - 1);
+      res->addr[0] = ADDRESS_VER_ED25519;  // Ed25519
+    } else {
+      printf("[%s:%d]: gets %s json addr failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      goto end;
+    }
+
+    free(addr);
+
+    // gets max_results
+    if ((ret = json_get_uint16(data_obj, key_maxResults, res->max_results)) != 0) {
+      printf("[%s:%d]: gets %s json max_results failed\n", __func__, __LINE__, key_maxResults);
+      ret = -1;
+      goto end;
+    }
+
+    res->max_results = (uint16_t)*number;
+
+    // gets count
+    if ((ret = json_get_uint16(data_obj, key_count, res->count)) != 0) {
+      printf("[%s:%d]: gets %s json count failed\n", __func__, __LINE__, key_count);
+      ret = -1;
+      goto end;
+    }
+
+    res->count = (uint16_t)*number;
+
+    // gets balance
+    if ((ret = json_get_uint64(data_obj, key_balance, res->balance)) != 0) {
+      printf("[%s:%d]: gets %s json balance failed\n", __func__, __LINE__, key_balance);
+      ret = -1;
+      goto end;
+    }
+
+    res->balance = (int64_t)*number;
+
+  end:
+    cJSON_Delete(json_obj);
+    return ret;
+  }
+}
+
+int get_balance(iota_client_conf_t const *conf, char *addr, res_balance_t *res) {
+  int ret = 0;
+  char const *const cmd_info = "api/v1/address/";
+
+  if (addr == NULL || res == NULL || conf == NULL) {
+    printf("[%s:%d]: get_balance failed (null parameter)\n", __func__, __LINE__);
+    return -1;
+  }
+
+  if (strlen(addr) != 64 && strlen(addr) != 81) {
+    printf("[%s:%d]: get_balance failed (invalid addr length)\n", __func__, __LINE__);
+    return -1;
+  }
+
+  // compose restful api command
+  iota_str_t *cmd = iota_str_new(conf->url);
+
+  if (iota_str_append(cmd, cmd_info)) {
+    printf("[%s:%d]: cmd_info append failed\n", __func__, __LINE__);
+    return -1;
+  }
+
+  if (iota_str_append(cmd, addr)) {
+    printf("[%s:%d]: addr append failed\n", __func__, __LINE__);
+    return -1;
+  }
+
+  // http client configuration
+  http_client_config_t http_conf = {0};
+  http_conf.url = cmd->buf;
+  if (conf->port) {
+    http_conf.port = conf->port;
+  }
+
+  byte_buf_t *http_res = byte_buf_new();
+  if (http_res == NULL) {
+    printf("[%s:%d]: OOM\n", __func__, __LINE__);
+    // TODO
```

Comment:
remove it

Reply:
@oopsmonk you mean remove `// TODO`, or something else?